src/plugins/chart_expressions/common/color_categories.ts

```diff
@@ -15,27 +15,38 @@ import { isMultiFieldKey } from '@kbn/data-plugin/common';
  */
 export function getColorCategories(
   rows: DatatableRow[],
-  accessor?: string
+  accessor?: string,
+  isTransposed?: boolean,
+  exclude?: any[]
 ): Array<string | string[]> {
-  return accessor
-    ? rows.reduce<{ keys: Set<string>; categories: Array<string | string[]> }>(
-        (acc, r) => {
-          const value = r[accessor];
-          if (value === undefined) {
-            return acc;
-          }
+  const ids = isTransposed
```

Comment:
Maybe this logic can be pushed upstream to the caller?

Reply:
I thought about doing that but I landed on putting it here to reduce code duplication.